pkg/webhook/server/certificate/manager.go

```diff
@@ -55,123 +54,152 @@ const (
 // It will also update the webhook caBundle field with the cluster CA cert and
 // approve the generated cert/key with k8s certification approval mechanism
 func NewManager(
-	crMgr crmanager.Manager,
+	client client.Client,
 	webhookName string,
 	webhookType WebhookType,
-	certDir string, certFileName string, keyFileName string, caConfigMapKey types.NamespacedName, caConfigMapField string) (*manager, error) {
-
-	certStore, err := NewFilePairStore(
-		certDir,
-		certFileName,
-		keyFileName,
-	)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to initialize new webhook cert/key file store")
-	}
+) *manager {
 
 	m := &manager{
-		crMgr:            crMgr,
-		log:              logf.Log.WithName("webhook/server/certificate/manager"),
-		certStore:        certStore,
-		caConfigMapKey:   caConfigMapKey,
-		caConfigMapField: caConfigMapField,
+		stopCh:           make(chan struct{}),
+		client:           client,
+		webhookName:      webhookName,
+		webhookType:      webhookType,
+		keyPairByService: map[types.NamespacedName]*triple.KeyPair{},
+		now:              time.Now,
+		ready:            false,
+		log: logf.Log.WithName("webhook/server/certificate/manager").
+			WithValues("webhookType", webhookType, "webhookName", webhookName),
 	}
+	return m
+}
 
-	dnsNames := []string{}
-	if webhookType == MutatingWebhook {
-		mutatingWebHookConfig, err := m.updateMutatingWebhookCABundle(webhookName)
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to update CA bundle at webhook")
+// Will start the the underlaying client-go cert manager [1]  and
+// wait for TLS key and cert to be generated
+//
+// [1] https://godoc.org/k8s.io/client-go/util/certificate
+func (m *manager) Start() error {
+	m.log.Info("Starting cert manager")
+
+	go wait.Until(func() {
+		deadline := m.nextRotationDeadline()
+		if sleepInterval := deadline.Sub(m.now()); sleepInterval > 0 {
```

Comment:
We use `m.now()` so we can mock `now`  so cannot use `time.Until` since it's uses directly `time.Now()` also After uses `Time` not `Duration` and `time.Until` returns `Duration`, but I really like the `Until` idiom let's add a method to the manager to has the idiom and unit test mocking.